### PR TITLE
Rebase of changes from branch minigame2-polish from old PR (JSON Question format and have started tutorial with question and logic #80 Implement context popup for minigame 2 (Tutorial + Explanation) #96)

### PR DESCRIPTION
**Pull request by Maxi-Garcia**
- Made changes from past PRs that modified these files
- Could not do a rebase as there was some logical conflicts so I had to make the changes on a new file
- Will merge these changes and request same people to review

Taking Description from closed PR as the changes are identical but only conform to dev branch latest updates:

-Started working on implementing logic + testing on mini-game 2 for a tutorial
-New helper functions added for testing all logic in the budget-game and questions logic files to allow jumping to different questions to test extensively
-Finalizing tutorial content and will check during stand-up on idea for tutorial and progress

### CODE REVIEW

**Review by michelemurakami — APPROVED** [review]
LGTM!

File: src/minigames/minigame2-budgeting/budget-game-setup.ts [hunk 1]
```diff
@@ -24,21 +24,40 @@ export const useBudgetGameLogic = (progressApi?: ProgressApi) => {
...
+      setContent("You got this question wrong... Try flipping days to be the exact value!")
+      //stop function early
+      return;
+    }
+
+    /* MINIGAME SECTION */
 
     if (isCorrect){
       setTitle("Correct!");
       setContent("You got this question right!")
```
Comment (fnriv) [inline src/minigames/minigame2-budgeting/budget-game-setup.ts]: I would change the tutorial correct description to say that it's the end of the tutorial.

File: src/minigames/minigame2-budgeting/budget-game-setup.ts [hunk 2]
```diff
@@ -48,9 +67,9 @@ export const useBudgetGameLogic = (progressApi?: ProgressApi) => {
     nextQuestion();
     resetButtons();
     //add end of screen pop-up here
-  if ( (currentQuestion.id) === questionCount){
+  if ( (currentQuestion.id) === questionCount - 1){
```
Comment (fnriv) [inline src/minigames/minigame2-budgeting/budget-game-setup.ts]: When testing this manually, there is no correct/incorrect popup for the last question; it goes straight into the end of screen popup.

**Review by fnriv — DISMISSED** [review]
Good work! I could run locally and all tests passed. You also have great code design (it was helpful when I was coding too. 

I've left a couple of quick changes regarding descriptions and an edge case for the questions:

- [ ] For the tutorial, I was a bit confused on when the tutorial ended and when the regular questions started. This should be quick to fix with more descriptive/specific tutorial descriptions.
- [ ] During the regular questions, on the last question, instead of the normal correct/incorrect popup, the page switches to the game over popup. I think for continuity, it would be good to ensure that the last question gets a correct/incorrect popup.

Other than that, LGTM. Thanks!

File: src/minigames/minigame2-budgeting/budget-game-setup.ts (hunk 2 above)
Comment (Maxi-Garcia) [reply src/minigames/minigame2-budgeting/budget-game-setup.ts]: had to refactor a lot of the game logic to merge the final game over pop to include information about the final question naturally. ran into some issues trying to format it though, but it includes both now.

File: src/minigames/minigame2-budgeting/budget-game-setup.ts (hunk 1 above)
Comment (Maxi-Garcia) [reply src/minigames/minigame2-budgeting/budget-game-setup.ts]: Addressed and included now in message creation

**Review by fnriv — COMMENTED** [review]
LGTM! We discussed the edge case of the correct/incorrect popup showing on the last question, and given the scope of our sprint, I believe that bug is not a priority right now as the main game navigation still works.
